### DISCUSSION TITLE
Fetching a many-to-one DTO projection

### DISCUSSION THREAD

**Discussion by bacca87** [micronaut-projects/micronaut-data#1614] [General]
### Issue description

Assuming we have 2 entities:

```
@MappedEntity
@Data
public class Book {
        @Id
        @GeneratedValue
        private int id;

        @NotNull
        private String title;

        @Relation(Relation.Kind.MANY_TO_ONE)
        private Author author;
}
```

```
@MappedEntity
@Data
public class Author {
    @GeneratedValue
    @Id
    private Long id;
    private final String name;
}
```

and the DTO:

```
@Introspected
public class AuthorDTO {
    private Long id;
    private final String name;
    private Set<int> books;
}
```

Is it possible to fetch the following query:

```
SELECT a.id, a.name, b.id as books
FROM author a 
INNER JOIN book b on b.author_id = a.id
```

results:
| id | name | books |
|-|-|-|
| 1    | Stephen King | 1  |
| 1    | Stephen King | 2  |
| 1    | Stephen King | 3  |

to the AuthorDTO creating only a single instance with a book list instead of creating 3 different DTOs?

**Comment by dstepanov** [comment]
At this moment DTOs don't support relations. You can try to annotate `AuthorDTO` with `@MappedEntity` that should use an ordinary entity mapper.

**Reply by bacca87** [reply]
I tried your workaround and it works! 
I hope DTO relations will be supported soon...
